engines/xeen/resources.h

```diff
@@ -463,6 +473,117 @@ class Resources {
 	const char *DARKSIDE_ENDING2;
 	const char *PHAROAH_ENDING_TEXT1;
 	const char *PHAROAH_ENDING_TEXT2;
+
+	struct {
+
+		struct {
+			int KEY_ITEM;
+			int KEY_QUICK;
+			int KEY_EXCHANGE;
+			int KEY_ESCAPE;
+		} DIALOGS_CHAR_INFO;
+
+		struct {
+			int KEY_FXON;
+			int KEY_MUSICON;
+			int KEY_LOAD;
+			int KEY_SAVE;
+			int KEY_QUIT;
+			int KEY_MRWIZARD;
+		} DIALOGS_CONTROL_PANEL;
+
+		struct {
+			int KEY_ROLL;
+			int KEY_CREATE;
+			int KEY_MGT;
+			int KEY_INT;
+			int KEY_PER;
+			int KEY_END;
+			int KEY_SPD;
+			int KEY_ACY;
+			int KEY_LCK;
+		} DIALOGS_CREATE_CHAR;
+
+		struct {
+			int KEY_ADVENTURER;
+			int KEY_WARRIOR;
+		} DIALOGS_DIFFICULTY;
+
+		struct {
+			int KEY_WEAPONS;
+			int KEY_ARMOR;
+			int KEY_ACCESSORY;
+			int KEY_MISC;
+			int KEY_ENCHANT;
+			int KEY_USE;
+			int KEY_BUY;
+			int KEY_SELL;
+			int KEY_IDENTIFY;
+			int KEY_FIX;
+			int KEY_EQUIP;
+			int KEY_REM;
+			int KEY_DISC;
+			int KEY_QUEST;
+			int KEY_RECHRG;
+			int KEY_GOLD;
+		} DIALOGS_ITEMS;
+
+		struct {
+			int KEY_DELETE;
+			int KEY_REMOVE;
+			int KEY_CREATE;
+			int KEY_EXIT;
+		} DIALOGS_PARTY;
+
+		struct {
+			int KEY_QUEST_ITEMS;
+			int KEY_CURRENT_QUESTS;
+			int KEY_AUTO_NOTES;
+		} DIALOGS_QUESTS;
+
+		struct {
+			int KEY_NEXT;
+		} DIALOGS_QUICK_FIGHT;
+
+		struct {
+			int KEY_CAST;
+			int KEY_NEW;
+			int KEY_FIRE;
+			int KEY_ELEC;
+			int KEY_COLD;
+			int KEY_ACID;
+			int KEY_SET;
+			int KEY_RETURN;
+		} DIALOGS_SPELLS;
+
+		struct {
+			int KEY_DEP;
+			int KEY_WITH;
+			int KEY_GOLD;
+			int KEY_GEMS;
+			int KEY_BROWSE;
+			int KEY_BUY_SPELLS;
+			int KEY_SPELL_INFO;
+			int KEY_SIGN_IN;
+			int KEY_DRINK;
+			int KEY_FOOD;
+			int KEY_TIP;
+			int KEY_RUMORS;
+			int KEY_HEAL;
+			int KEY_DONATION;
+			int KEY_UNCURSE;
+			int KEY_TRAIN;
+		} LOCATIONS;
+
+		struct {
+			int KEY_START_NEW_GAME;
+			int KEY_LOAD_GAME;
+			int KEY_SHOW_CREDITS;
+			int KEY_VIEW_ENDGAME;
+		} CLOUDSOFXEENMENU;
+	} KEY_CONSTANTS;
```

Comment:
This structure is not well named as it's not constant at all.
All caps names are reserved for constants, which it's not.
It confused me when reading the switch to if conversion above (which makes sense when variable is not constant)

Reply:
Fixed